Automattic/jetpack
# Social Menu / Social Icons Widget: add Mastodon

Fixes #10338

#### Changes proposed in this Pull Request:

This is a first implementation, that we can revisit once #28172 is merged.

For now, it detects some of the most popular instances of Mastodon (pulled from https://instances.social/ plus a few others that I know are popular in the WordPress community) and uses the Mastodon icon there.

It's not perfect since we cannot possibly detect all instances of Mastodon, but that's a start.

<img width="145" alt="Screenshot 2023-01-04 at 19 08 11" src="https://user-images.githubusercontent.com/426388/210622098-f1230811-da13-4356-aa2f-a17fa2b5b6ae.png">

<img width="392" alt="Screenshot 2023-01-04 at 19 08 07" src="https://user-images.githubusercontent.com/426388/210622108-c4550826-bd30-492a-b4e5-799b599a1eaa.png">


#### Other information:


#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

List of sample links to use:

```
https://mastodon.online/@kraft
https://facebook.com/jetpackme
https://religious.social/@kraft/
https://mastodon.jetpack.com/@jetpack
https://www.amazon.fr/Lucky-Duck-Games-Jetpack-Fran%C3%A7aise/dp/B07SVDFMCD/
https://www.google.it/maps
https://telegram.me/jeherve
https://profiles.wordpress.org/jeherve/
```

* Start by switching to an old theme, like Twenty Ten.
* Go to Plugins > Add New and install the Classic Widgets plugin.
* In a functionality plugin, add the following:
```php
// Add social menu support
add_action( 'after_setup_theme', function () {
	add_theme_support( 'jetpack-social-menu', 'svg' );
} );
// Add an extra instance to the list.
add_filter(
	'jetpack_mastodon_instance_list',
	function ( $mastodon_instance_list ) {
		$mastodon_instance_list[] = 'religious.social';
		return $mastodon_instance_list;
	}
);
```
* Go to Appearance > Customize > Menus
* Create a new menu with custom links from the example list above.
* Assign that menu to the Social Menu location
* Go to Appearance > Widgets menu
* Drag the Social Icons widget to one of the active widget areas, and add some links again.
* Check how things look like on the frontend.
